File: demo-service-impl/src/main/java/ru/zubmike/service/demo/dao/StarshipDao.java
package ru.zubmike.service.demo.dao;

import ru.zubmike.core.dao.EntityItemDao;
import ru.zubmike.service.demo.types.Starship;

import java.util.Optional;

public interface StarshipDao extends EntityItemDao<Long, Starship> {

	Optional<Starship> getByNumber(String number);

}

File: demo-service-kotlin-impl/src/main/kotlin/ru/zubmike/service/demo/dao/StarshipDao.kt
package ru.zubmike.service.demo.dao

import ru.zubmike.core.dao.EntityItemDao
import ru.zubmike.service.demo.types.Starship

interface StarshipDao : EntityItemDao<Long, Starship> {

    fun getByNumber(number: String): Starship?

}
